androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/MapAndSheetPage.kt

```diff
@@ -423,189 +437,235 @@ fun MapAndSheetPage(
+                enterTransition = {
+                    val initialRoute = SheetRoutes.fromNavBackStackEntry(this.initialState)
+                    val (initialStopId, initialStopFilter) =
+                        when (initialRoute) {
+                            is SheetRoutes.StopDetails ->
+                                Pair(initialRoute.stopId, initialRoute.stopFilter)
+
+                            else -> Pair(null, null)
+                        }
 
-                SheetPage {
-                    FavoritesPage(
-                        openSheetRoute = ::navigate,
-                        favoritesViewModel = favoritesViewModel,
-                        errorBannerViewModel = errorBannerViewModel,
-                        nearbyTransit = nearbyTransit,
-                    )
-                }
-            }
-            composable<SheetRoutes.EditFavorites>(typeMap = SheetRoutes.typeMap) {
-                SheetPage {
-                    EditFavoritesPage(
-                        global = nearbyTransit.globalResponse,
-                        favoritesViewModel = favoritesViewModel,
-                        onClose = {
-                            navController.popBackStackFrom(SheetRoutes.EditFavorites::class)
-                        },
-                    )
-                }
-            }
-            composable<SheetRoutes.StopDetails>(typeMap = SheetRoutes.typeMap) { backStackEntry ->
-                val navRoute: SheetRoutes.StopDetails = backStackEntry.toRoute()
-                val filters =
-                    StopDetailsPageFilters(
-                        navRoute.stopId,
-                        navRoute.stopFilter,
-                        navRoute.tripFilter,
-                    )
+                    val targetRoute = SheetRoutes.fromNavBackStackEntry(this.targetState)
+                    val (targetStopId, targetStopFilter) =
+                        when (targetRoute) {
+                            is SheetRoutes.StopDetails ->
+                                Pair(targetRoute.stopId, targetRoute.stopFilter)
+
+                            else -> Pair(null, null)
+                        }
 
-                LaunchedEffect(navRoute) {
-                    if (navBarVisible) {
-                        hideNavBar()
+                    // Skip animation if navigating within a single stop
+                    if (
+                        initialStopId != null &&
+                            targetStopId != null &&
+                            initialStopId == targetStopId &&
+                            initialStopFilter?.routeId == targetStopFilter?.routeId
+                    ) {
+                        EnterTransition.None
                     }
-                    if (filters.stopFilter != null) {
-                        analytics.track(AnalyticsScreen.StopDetailsFiltered)
+
+                    if (initialRoute == targetRoute) {
+                        EnterTransition.None
```

Comment:
Added to reduce jitter. Otherwise, nearby page was sliding in after the loading tab state, which was jarring.